pkg/network/http/monitor.go

```diff
@@ -165,10 +181,33 @@ func (m *Monitor) GetHTTPStats() map[Key]RequestStats {
 		return nil
 	}
 
-	reply := make(chan map[Key]RequestStats, 1)
+	reply := make(chan HTTPMonitorStats, 1)
 	defer close(reply)
 	m.pollRequests <- reply
-	return <-reply
+	stats := <-reply
+	m.telemetrySnapshot = &stats.telemetry
+	return stats.requestStats
+}
+
+func (m *Monitor) GetStats() map[string]int64 {
+	if m == nil {
+		return map[string]int64{}
+	}
+
+	m.mux.Lock()
+	defer m.mux.Unlock()
+	if m.stopped {
+		return map[string]int64{}
+	}
+
+	if m.telemetrySnapshot == nil {
+		return map[string]int64{}
```

Comment:
All the cases doing `return map[string]int64{}` can do `return nil` actually, since reading from a `nil` map is OK.